add Gizmo Engineering GK6 keyboard

federicoweber opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description
This add the [Gizmo Engineering GK6](https://gizmo.engineering/) keyboard.
![GK6](https://i.imgur.com/M9xxMzn.jpg)


Keyboard Maintainer: [federicoweber](https://github.com/federicoweber)
Hardware Supported: atmega32a, Gizmo Engineering GK6
Hardware Availability: [Gizmo Engineering](https://gizmo.engineering/)

<!--- Describe your changes in detail here. -->

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on keyboards/gizmo_engineering/gk6/config.h:
```diff
@@ -0,0 +1,80 @@
+#define DIODE_DIRECTION ROW2COL
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 5
```
```suggestion
#define DEBOUNCE 5
```

drashna commented on keyboards/gizmo_engineering/gk6/config.h:
```diff
@@ -0,0 +1,80 @@
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
```
```suggestion
```

drashna commented on keyboards/gizmo_engineering/gk6/config.h:
```diff
@@ -0,0 +1,80 @@
+#define DISABLE_RGB_MATRIX_BAND_SAT
+#define DISABLE_GB_MATRIX_BAND_PINWHEEL_SAT
+#define DISABLE_RGB_MATRIX_BAND_SPIRAL_SAT
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true
```
Not needed twice. 😄 
```suggestion
```

drashna commented on keyboards/gizmo_engineering/gk6/gk6.c:
```diff
@@ -0,0 +1,26 @@
+ */
+#include "gk6.h"
+
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
No longer needed. 
```suggestion
```

drashna commented on keyboards/gizmo_engineering/gk6/gk6.h:
```diff
@@ -0,0 +1,63 @@
+    { k40, k41, k42, k43, k44, KC_NO, k45, k47, k48, k49, k4a, k4b }, \
+}
+
+#define LAYOUT_gk6_5x12( \
```
```suggestion
#define LAYOUT_ortho_5x12( \
```

drashna commented on keyboards/gizmo_engineering/gk6/info.json:
```diff
@@ -0,0 +1,211 @@
+                {"x": 11, "y": 4}
+            ]
+        },
+        "LAYOUT_gk6_5x12": {
```
```suggestion
        "LAYOUT_ortho_5x12": {
```

drashna commented on keyboards/gizmo_engineering/gk6/keymaps/12x5/keymap.c:
```diff
@@ -0,0 +1,261 @@
+
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+        case LOWER:
+          if (record->event.pressed) {
+            layer_on(_LOWER);
+          } else {
+            layer_off(_LOWER);
+          }
+          return false;
+          break;
+        case RAISE:
+          if (record->event.pressed) {
+            layer_on(_RAISE);
+          } else {
+            layer_off(_RAISE);
+          }
+          return false;
+          break;
+      }
+    return true;
+};
```
These all work as `MO`
```suggestion
```

federicoweber replied:
ah nice, much cleaner 👍

drashna commented on keyboards/gizmo_engineering/gk6/keymaps/12x5/keymap.c:
```diff
@@ -0,0 +1,261 @@
+  _RAISE,
+};
+
+enum keycodes {
+  LOWER,
+  RAISE,
+};
```
```suggestion
#define LOWER MO(_LOWER)
#define RAISE MO(_RAISE)
```

drashna commented on keyboards/gizmo_engineering/gk6/keymaps/12x5/keymap.c:
```diff
@@ -0,0 +1,261 @@
+    return true;
+};
+
+const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```
This should go in the keyboard's c file, not here

federicoweber replied:
Good one @drashna, the reason I've put this in here is they are different for each layout, as we don't LEDs without switches to turn on.

drashna replied:
Yeah, it should be in the keyboard's C file, no matter what. 

You could use `rgb_matrix_indicator_user` to effectively disable the LEDs in question. 
Or create revisions/subfolders for the different layouts.

drashna commented on keyboards/gizmo_engineering/gk6/keymaps/test/rules.mk:
```diff
@@ -0,0 +1,38 @@
+# MCU name
+#MCU = at90usb1286
+MCU = atmega32u4
+
+# Processor frequency.
+F_CPU = 16000000
+
+
+# Target architecture (see library "Board Types" documentation).
+ARCH = AVR8
+
+# Input clock frequency.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
```
```suggestion
MCU = atmega32u4

```

drashna commented on keyboards/gizmo_engineering/gk6/rules.mk:
```diff
@@ -0,0 +1,39 @@
+# MCU name
+#MCU = at90usb1286
+MCU = atmega32u4
+
+# Processor frequency.
+F_CPU = 16000000
+
+
+# Target architecture (see library "Board Types" documentation).
+ARCH = AVR8
+
+# Input clock frequency.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
```
```suggestion
# MCU name
MCU = atmega32u4

```

federicoweber commented:
Thanks for the review @drashna, I've addressed your changes request and left a comment on the reason why the led matrix configuration is in the keymap.

federicoweber commented:
Hey @drashna, sorry for the delay, it should be all updated now.
Please let me know if there is anything else I need to change.

Thanks for your help!

zvecr commented on keyboards/gizmo_engineering/gk6/config.h:
```diff
@@ -0,0 +1,74 @@
+#define DISABLE_RGB_MATRIX_SPLASH
+#define DISABLE_RGB_MATRIX_SOLID_SPLASH
+#define DISABLE_RGB_MATRIX_BAND_SAT
+#define DISABLE_GB_MATRIX_BAND_PINWHEEL_SAT
```
```suggestion
#define DISABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
```

federicoweber replied:
ah great catch thanks!

zvecr commented on keyboards/gizmo_engineering/gk6/keymaps/12x5/config.h:
```diff
@@ -0,0 +1,20 @@
+/* Copyright 2021 Gizmo Engineering
```
This file should be removed

zvecr commented on keyboards/gizmo_engineering/gk6/keymaps/12x5/keymap.c:
```diff
@@ -0,0 +1,196 @@
+
+};
+
+void rgb_matrix_indicators_kb(void)
```
```suggestion
void rgb_matrix_indicators_user(void)
```

zvecr commented on keyboards/gizmo_engineering/gk6/keymaps/1x2u/config.h:
```diff
@@ -0,0 +1,20 @@
+/* Copyright 2021 Gizmo Engineering
```
This file should be removed

zvecr commented on keyboards/gizmo_engineering/gk6/keymaps/default/config.h:
```diff
@@ -0,0 +1,20 @@
+/* Copyright 2021 Gizmo Engineering
```
This file should be removed

zvecr commented on keyboards/gizmo_engineering/gk6/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,198 @@
+
+};
+
+void rgb_matrix_indicators_kb(void)
```
```suggestion
void rgb_matrix_indicators_user(void)
```

zvecr commented on keyboards/gizmo_engineering/gk6/keymaps/1x2u/keymap.c:
```diff
@@ -0,0 +1,197 @@
+
+};
+
+void rgb_matrix_indicators_kb(void)
```
```suggestion
void rgb_matrix_indicators_user(void)
```

zvecr commented on keyboards/gizmo_engineering/gk6/keymaps/test/keymap.c:
```diff
@@ -0,0 +1,201 @@
+  //debug_mouse=true;
+}
+
+void rgb_matrix_indicators_kb(void)
```
```suggestion
void rgb_matrix_indicators_user(void)
```

zvecr commented on keyboards/gizmo_engineering/gk6/keymaps/via/config.h:
```diff
@@ -0,0 +1,20 @@
+/* Copyright 2021 Gizmo Engineering
```
This file should be removed

zvecr commented on keyboards/gizmo_engineering/gk6/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,193 @@
+),
+
+};
+
+void rgb_matrix_indicators_kb(void)
+{
+}
+
```
```suggestion
```

zvecr commented on keyboards/gizmo_engineering/gk6/keymaps/test/rules.mk:
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+
+# Build Options
+#
+BOOTMAGIC_ENABLE = lite       # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = no        # Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
+CONSOLE_ENABLE = yes        # Console for debug(+400)
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = IS31FL3731
```
```suggestion
CONSOLE_ENABLE = yes
```

federicoweber commented:
Thanks for the review @zvecr, I've addressed all your change requests.

drashna commented on keyboards/gizmo_engineering/gk6/gk6.c:
```diff
@@ -0,0 +1,82 @@
+ */
+#include "gk6.h"
+
+const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```
```suggestion
#ifdef RGB_MATRIX_ENABLE
const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```

federicoweber replied:
Good call

drashna replied:
Also, again, will error out, but will be "fixed" when develop is merged in. 
```suggestion
const is31_led __flash g_is31_leds[DRIVER_LED_TOTAL] = {
```

drashna replied:
The `__flash` part is needed still.

drashna commented on keyboards/gizmo_engineering/gk6/gk6.c:
```diff
@@ -0,0 +1,82 @@
+    {1, C9_14, C8_14, C7_14},  //60 led58
+    {1, C9_15, C8_15, C6_14},  //61 led59
+    {1, C9_16, C7_15, C6_15},  //62 led60
+};
```
```suggestion
};
#endif
```

drashna commented:
My one concern here is that the LED layout exists only in the keymaps.  This means that if it's not included (such as compiled with the configurator), it will not work. And by not work, I mean that it simple will not compile. 

A better option may be to set a default version, and just modify it based on the layout.

federicoweber commented:
I hear you @drashna, the problem with that is that we can't override the LED layout if we defined it in the keyboard `c` file. And because the first part defines the correlation between switches and LEDs, it will change based on the layout.

You can see in here how the LEDs are physically organized, depending on the layout only a few of those will be on and mapped to a switch.

![20210820_114155](https://user-images.githubusercontent.com/992920/130279264-2916ab86-2e5e-4c4f-b183-967001dd3e8c.jpg)

Any suggestions on how to do that?

drashna commented:
In theory, since `g_led_config` isn't a const, you could add something like: 
```c
void keyboard_pre_init_user(void) {
    g_led_config.flags[43] = LED_FLAG_NONE;
}
```

This would effectively disable that LED, and allow for a single, unified LED config.

federicoweber commented:
Thanks for the pointers @drashna, I made it work by redefining the LEDs, to keys mapping with
```
void keyboard_pre_init_user(void) {
    //   5, 9, 14, 19, 24, 30, 36, 40, 46, 51, 55,  60
    g_led_config.matrix_co[4][0] = 5;
    g_led_config.matrix_co[4][1] = 9;
    g_led_config.matrix_co[4][2] = 14;
    g_led_config.matrix_co[4][3] = 19;
    g_led_config.matrix_co[4][4] = 24;
    g_led_config.matrix_co[4][5] = 30;
    g_led_config.matrix_co[4][6] = 36;
    g_led_config.matrix_co[4][7] = 41;
    g_led_config.matrix_co[4][8] = 47;
    g_led_config.matrix_co[4][9] = 52;
    g_led_config.matrix_co[4][10] = 57;
    g_led_config.matrix_co[4][11] = 62;
}
```

drashna commented on keyboards/gizmo_engineering/gk6/config.h:
```diff
@@ -0,0 +1,74 @@
+#define RGB_MATRIX_STARTUP_HUE 0
+#define RGB_MATRIX_STARTUP_SAT 255
+#define RGB_MATRIX_KEYPRESSES
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true
```
Technically, this will error out on `master` at this moment, but should be fixed when develop is merged (in less than a week). So: 
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED
```

drashna commented on keyboards/gizmo_engineering/gk6/keymaps/12x5/keymap.c:
```diff
@@ -0,0 +1,114 @@
+
+};
+
+void rgb_matrix_indicators_user(void)
+{
+    rgb_matrix_set_color(25, 0, 0, 0);
+    rgb_matrix_set_color(36, 0, 0, 0);
+    rgb_matrix_set_color(42, 0, 0, 0);
+}
+
+void keyboard_pre_init_user(void) {
+    // 5, 9,  14, 19, 24, 30, 35, 40, 46, 51, 55, 60
+    g_led_config.matrix_co[4][0] = 5;
+    g_led_config.matrix_co[4][1] = 9;
+    g_led_config.matrix_co[4][2] = 14;
+    g_led_config.matrix_co[4][3] = 19;
+    g_led_config.matrix_co[4][4] = 24;
+    g_led_config.matrix_co[4][5] = 30;
+    g_led_config.matrix_co[4][6] = 35;
+    g_led_config.matrix_co[4][7] = 41;
+    g_led_config.matrix_co[4][8] = 47;
+    g_led_config.matrix_co[4][9] = 52;
+    g_led_config.matrix_co[4][10] = 57;
+    g_led_config.matrix_co[4][11] = 62;
+}
```
```suggestion
#ifdef RGB_MATRIX_ENABLE
void rgb_matrix_indicators_user(void)
{
    rgb_matrix_set_color(25, 0, 0, 0);
    rgb_matrix_set_color(36, 0, 0, 0);
    rgb_matrix_set_color(42, 0, 0, 0);
}

void keyboard_pre_init_user(void) {
    // 5, 9,  14, 19, 24, 30, 35, 40, 46, 51, 55, 60
    g_led_config.matrix_co[4][0] = 5;
    g_led_config.matrix_co[4][1] = 9;
    g_led_config.matrix_co[4][2] = 14;
    g_led_config.matrix_co[4][3] = 19;
    g_led_config.matrix_co[4][4] = 24;
    g_led_config.matrix_co[4][5] = 30;
    g_led_config.matrix_co[4][6] = 35;
    g_led_config.matrix_co[4][7] = 41;
    g_led_config.matrix_co[4][8] = 47;
    g_led_config.matrix_co[4][9] = 52;
    g_led_config.matrix_co[4][10] = 57;
    g_led_config.matrix_co[4][11] = 62;
}
#endif
```

drashna commented on keyboards/gizmo_engineering/gk6/rgb_matrix_kb.inc:
```diff
@@ -0,0 +1,91 @@
+    4, 9, 14, 19, 24, 25, 30, 36, 35, 42, 41, 47, 52, 57, 62,
+};
+
+static void set_led_color(int i) {
```
```suggestion
static void set_led_color(uint8_t i) {
```

drashna reviewed (changes requested):
the if-defs should be added to all of the keymaps,

fauxpark commented on keyboards/gizmo_engineering/gk6/rules.mk:
```diff
@@ -0,0 +1,24 @@
+
+
+# Build Options
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = no        # Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
+CONSOLE_ENABLE = no         # Console for debug(+400)
+COMMAND_ENABLE = no         # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

fauxpark commented on keyboards/gizmo_engineering/gk6/rules.mk:
```diff
@@ -0,0 +1,24 @@
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
```suggestion
```

fauxpark commented on keyboards/gizmo_engineering/gk6/readme.md:
```diff
@@ -0,0 +1,14 @@
+# Gizmo Engineering GK6
+
+![GK6](https://i.imgur.com/M9xxMzn.jpg)
+
+
+Keyboard Maintainer: [federicoweber](https://github.com/federicoweber)
+Hardware Supported: atmega32a, Gizmo Engineering GK6
+Hardware Availability: [Gizmo Engineering](https://gizmo.engineering/)
```
```suggestion
![GK6](https://i.imgur.com/M9xxMznh.jpg)

* Keyboard Maintainer: [federicoweber](https://github.com/federicoweber)
* Hardware Supported: atmega32a, Gizmo Engineering GK6
* Hardware Availability: [Gizmo Engineering](https://gizmo.engineering/)
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,91 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+enum layers {
```
```suggestion

#include QMK_KEYBOARD_H

enum layers {
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,91 @@
+ * | Esc  | GUI  |  Alt |Lower |Space |Space |Space | Back |Raise | Left | Down |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_SLSH, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,  \
+  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_BSPC, KC_BSPC, KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_RGHT  \
```
```suggestion
[_QWERTY] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_SLSH,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,
  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_BSPC, KC_BSPC, KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_RGHT
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,91 @@
+ * |      |      |      |      |      |      |      |      |      |      |      |R Alt |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ortho_5x12( \
+  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, _______,  _______, _______, _______, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, _______, \
+  _______, _______,  _______, _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, _______, \
+  _______, RGB_TOG,  RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, _______, _______, _______, KC_PIPE, KC_BSLS, \
+  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_RALT  \
```
```suggestion
[_RAISE] = LAYOUT_ortho_5x12(
  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, _______,  _______, _______, _______, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, _______,
  _______, _______,  _______, _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, _______,
  _______, RGB_TOG,  RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, _______, _______, _______, KC_PIPE, KC_BSLS,
  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_RALT
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,91 @@
+ * |      |      |      |      |      |      |      |      |      |      | PgDn |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ortho_5x12( \
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+  KC_CAPS, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, KC_HOME, KC_END,  KC_INS,  KC_MPLY, _______, _______, _______, _______, _______, _______, _______, \
+  _______, KC_MPRV, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______, _______, _______, KC_PGUP, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN, _______  \
```
```suggestion
[_LOWER] = LAYOUT_ortho_5x12(
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
  KC_CAPS, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, KC_HOME, KC_END,  KC_INS,  KC_MPLY, _______, _______, _______, _______, _______, _______, _______,
  _______, KC_MPRV, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______, _______, _______, KC_PGUP, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN, _______
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/test/rules.mk:
```diff
@@ -0,0 +1 @@
+CONSOLE_ENABLE = yes        # Console for debug(+400)
```
```suggestion
CONSOLE_ENABLE = yes
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/test/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |      |      |      |      |      |      |      |      |      |      | PgDn |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ortho_5x12( \
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+  KC_CAPS, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, KC_HOME, KC_END,  KC_INS,  KC_MPLY, _______, _______, _______, _______, _______, _______, _______, \
+  _______, KC_MPRV, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______, _______, _______, KC_PGUP, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN, _______  \
```
```suggestion
[_LOWER] = LAYOUT_ortho_5x12(
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
  KC_CAPS, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, KC_HOME, KC_END,  KC_INS,  KC_MPLY, _______, _______, _______, _______, _______, _______, _______,
  _______, KC_MPRV, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______, _______, _______, KC_PGUP, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN, _______
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/test/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |EEP_R |      |      |      |      |      |      |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ortho_5x12( \
+  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, _______,  _______, _______, _______, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, _______, \
+  _______, _______,  _______, _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, _______, \
+  _______, RGB_TOG,  RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, _______, _______, _______, KC_PIPE, KC_BSLS, \
+  EEP_RST,   _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
```
```suggestion
[_RAISE] = LAYOUT_ortho_5x12(
  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, _______,  _______, _______, _______, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, _______,
  _______, _______,  _______, _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, _______,
  _______, RGB_TOG,  RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, _______, _______, _______, KC_PIPE, KC_BSLS,
  EEP_RST,   _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/test/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * | CTRL | GUI  |  Alt |Lower |  Back|Space |Space |  Alt |Raise | Left | Down |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_SLSH, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,  \
+  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_BSPC, KC_SPC,  KC_SPC,  KC_RALT, RAISE,   KC_LEFT, KC_DOWN, KC_RGHT  \
```
```suggestion
[_QWERTY] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_SLSH,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,
  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_BSPC, KC_SPC,  KC_SPC,  KC_RALT, RAISE,   KC_LEFT, KC_DOWN, KC_RGHT
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,102 @@
+ * |      |      |      |      |             |             |      |      | PgDn |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_gk6_2x2u( \
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+  KC_CAPS, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, KC_HOME, KC_END,  KC_INS,  KC_MPLY, _______, _______, _______, _______, _______, _______, _______, \
+  _______, KC_MPRV, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______, _______, _______, KC_PGUP, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN, _______  \
```
```suggestion
[_LOWER] = LAYOUT_gk6_2x2u(
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
  KC_CAPS, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, KC_HOME, KC_END,  KC_INS,  KC_MPLY, _______, _______, _______, _______, _______, _______, _______,
  _______, KC_MPRV, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______, _______, _______, KC_PGUP, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN, _______
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,102 @@
+ * |Debug |      |      |      |             |             |      |      |      |R Alt |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_gk6_2x2u( \
+  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, _______,  _______, _______, _______, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, _______, \
+  _______, _______,  _______, _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, _______, \
+  _______, RGB_TOG,  RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, _______, _______, _______, KC_PIPE, KC_BSLS, \
+  DEBUG,   _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_RALT  \
```
```suggestion
[_RAISE] = LAYOUT_gk6_2x2u(
  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, _______,  _______, _______, _______, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, _______,
  _______, _______,  _______, _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, _______,
  _______, RGB_TOG,  RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, _______, _______, _______, KC_PIPE, KC_BSLS,
  DEBUG,   _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_RALT
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,102 @@
+ * | Esc  | GUI  |  Alt |Lower |  Backspace  |    Space    |Raise | Left | Down |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_gk6_2x2u( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_SLSH, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,  \
+  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_BSPC, KC_BSPC, KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_RGHT  \
```
```suggestion
[_QWERTY] = LAYOUT_gk6_2x2u(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_SLSH,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,
  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_BSPC, KC_BSPC, KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_RGHT
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,102 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+enum layers {
```
```suggestion

#include QMK_KEYBOARD_H

enum layers {
```

fauxpark replied:
Still outstanding.

federicoweber replied:
oh sorry I missed this one

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/1x2u/keymap.c:
```diff
@@ -0,0 +1,117 @@
+ * |      |      |      |      |      |             |      |      |      | PgDn |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_gk6_1x2u( \
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+  KC_CAPS, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, KC_HOME, KC_END,  KC_INS,  KC_MPLY, _______, _______, _______, _______, _______, _______, _______, \
+  _______, KC_MPRV, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______, _______, _______, KC_PGUP, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN, _______  \
```
```suggestion
[_LOWER] = LAYOUT_gk6_1x2u(
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
  KC_CAPS, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, KC_HOME, KC_END,  KC_INS,  KC_MPLY, _______, _______, _______, _______, _______, _______, _______,
  _______, KC_MPRV, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______, _______, _______, KC_PGUP, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN, _______
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/1x2u/keymap.c:
```diff
@@ -0,0 +1,117 @@
+ * |Debug |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_gk6_1x2u( \
+  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, _______,  _______, _______, _______, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, _______, \
+  _______, _______,  _______, _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, _______, \
+  _______, RGB_TOG,  RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, _______, _______, _______, KC_PIPE, KC_BSLS, \
+  DEBUG,   _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
```
```suggestion
[_RAISE] = LAYOUT_gk6_1x2u(
  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, _______,  _______, _______, _______, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, _______,
  _______, _______,  _______, _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, _______,
  _______, RGB_TOG,  RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, _______, _______, _______, KC_PIPE, KC_BSLS,
  DEBUG,   _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/1x2u/keymap.c:
```diff
@@ -0,0 +1,117 @@
+ * | Esc  | GUI  |  Alt |Lower |  Back|    Space    |  Alt |Raise | Left | Down |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_gk6_1x2u( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_SLSH, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,  \
+  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_BSPC, KC_SPC, KC_SPC,  KC_RALT,  RAISE,   KC_LEFT, KC_DOWN, KC_RGHT  \
```
```suggestion
[_QWERTY] = LAYOUT_gk6_1x2u(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_SLSH,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,
  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_BSPC, KC_SPC, KC_SPC,  KC_RALT,  RAISE,   KC_LEFT, KC_DOWN, KC_RGHT
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/1x2u/keymap.c:
```diff
@@ -0,0 +1,117 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+enum layers {
```
```suggestion

#include QMK_KEYBOARD_H

enum layers {
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/12x5/keymap.c:
```diff
@@ -0,0 +1,116 @@
+ * |      |      |      |      |      |      |      |      |      |      | PgDn |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ortho_5x12( \
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+  KC_CAPS, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, KC_HOME, KC_END,  KC_INS,  KC_MPLY, _______, _______, _______, _______, _______, _______, _______, \
+  _______, KC_MPRV, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______, _______, _______, KC_PGUP, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN, _______  \
```
```suggestion
[_LOWER] = LAYOUT_ortho_5x12(
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
  KC_CAPS, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, KC_HOME, KC_END,  KC_INS,  KC_MPLY, _______, _______, _______, _______, _______, _______, _______,
  _______, KC_MPRV, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______, _______, _______, KC_PGUP, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN, _______
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/12x5/keymap.c:
```diff
@@ -0,0 +1,116 @@
+ * |Debug |      |      |      |      |      |      |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ortho_5x12( \
+  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, _______,  _______, _______, _______, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, _______, \
+  _______, _______,  _______, _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, _______, \
+  _______, RGB_TOG,  RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, _______, _______, _______, KC_PIPE, KC_BSLS, \
+  DEBUG,   _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
```
```suggestion
[_RAISE] = LAYOUT_ortho_5x12(
  _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, _______,  _______, _______, _______, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, _______,
  _______, _______,  _______, _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, _______,
  _______, RGB_TOG,  RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, _______, _______, _______, KC_PIPE, KC_BSLS,
  DEBUG,   _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/12x5/keymap.c:
```diff
@@ -0,0 +1,116 @@
+ * | Esc  | GUI  |  Alt |Lower |  Back|Space |Space |  Alt |Raise | Left | Down |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_SLSH, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,  \
+  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_BSPC, KC_SPC, KC_SPC,  KC_RALT,  RAISE,   KC_LEFT, KC_DOWN, KC_RGHT  \
```
```suggestion
[_QWERTY] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_SLSH,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,
  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_BSPC, KC_SPC, KC_SPC,  KC_RALT,  RAISE,   KC_LEFT, KC_DOWN, KC_RGHT
```

fauxpark commented on keyboards/gizmo_engineering/gk6/gk6.h:
```diff
@@ -0,0 +1,63 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
```
```suggestion

#pragma once
```

fauxpark commented on keyboards/gizmo_engineering/gk6/gk6.c:
```diff
@@ -0,0 +1,182 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "gk6.h"
```
```suggestion

#include "gk6.h"
```

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/12x5/keymap.c:
```diff
@@ -0,0 +1,116 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+enum layers {
```
```suggestion

#include QMK_KEYBOARD_H

enum layers {
```

federicoweber commented:
Thanks @fauxpark

drashna commented on keyboards/gizmo_engineering/gk6/gk6.c:
```diff
@@ -0,0 +1,183 @@
+    {224, 64}, // 62
+}, {
+    // LED Index to Flag
+    0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
```
These should be set to match the type of keys that they're associated with

federicoweber commented:
Hi @drashna, and @fauxpark.
Sorry for the long silence on this one.
I went ahead and rebased my branch against the latest `master` branch, and also addressed the last few pending comments.

Please let me know if there is anything else you want me to update before merging.

Thanks!

fauxpark commented on keyboards/gizmo_engineering/gk6/keymaps/test/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+enum layers {
```
```suggestion

#include QMK_KEYBOARD_H

enum layers {
```

fauxpark commented on keyboards/gizmo_engineering/gk6/rgb_matrix_kb.inc:
```diff
@@ -0,0 +1,91 @@
+#include "print.h"
+
```
```suggestion
```

fauxpark commented on keyboards/gizmo_engineering/gk6/rules.mk:
```diff
@@ -0,0 +1,24 @@
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

federicoweber commented:
hey @fauxpark is there anything blocking left to merge this in?

drashna commented on keyboards/gizmo_engineering/gk6/config.h:
```diff
@@ -0,0 +1,66 @@
+#define DRIVER_1_LED_TOTAL 32
+#define DRIVER_2_LED_TOTAL 32
+#define DRIVER_LED_TOTAL (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+#define RGBLIGHT_EFFECT_BREATHE_MAX 250  // 0-255
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 250
+#define RGB_MATRIX_STARTUP_HUE 0
+#define RGB_MATRIX_STARTUP_SAT 255
```
These aren't really needed, as they're pretty much the default.

federicoweber replied:
ah make sense, just pushed the change.
Thanks!

federicoweber commented:
hey @drashna, is there any blocker to merge this in?

Thanks!

federicoweber commented:
Hi @fauxpark sorry, somehow I completely missed those.
I've committed your suggestions.

Thanks!

federicoweber commented:
Thanks for merging this in @drashna!
